### PR TITLE
Warn on missing "Change log entry" section

### PR DESCRIPTION
**Pull request by Strech**
**What does this PR do?**

Adds new PR check which will notify either author or DataDog/ruby-guild about missing "Change log entry" section.

**Motivation:**

Automate work which could be automated, less personal, more efficient.

**Change log entry**

None.

**Additional Notes:**

It will trigger only on PR open/re-open/edit events which makes sense. After "Change log entry" is filled in it will update its comment with a "Thank You" message.

Warning message:
Screenshot 2024-12-04 at 12 23 46

Thank you message:
Screenshot 2024-12-04 at 12 25 28

_P.S This PR is using regex to solve the problem_ :trollface:

**How to test the change?**

Create a branch based on this and open PR with incomplete "Change log entry" section. Alternatively `act` from Github.

### CODE REVIEW

**Comment by Strech** [comment]
Please fill CHANGELOG entry section.

      If your changes need mention in a CHANGELOG
      > Yes. <brief summary to be placed into the change log>

      or as follows if they don't
      > No.

**Comment by Strech** [comment]
@p-datadog I'm working on it, it will be a non-blocking check which will write a message in PR depending on many factors:

1. Is it community PR
2. Is the message in place
3. Does it follow the format (Yes/No/ ...)

**Comment by Strech** [comment]
:wave: Please fill "Change log entry" section in the pull request description.

    If your changes need to be present in [CHANGELOG.md](https://github.com/DataDog/dd-trace-rb/blob/master/CHANGELOG.md) you can state it this way

    ```md
    **Change log entry**

    Yes. A brief summary to be placed into the CHANGELOG.md
    ```
    _(possible answers Yes/Yep/Yeah)_

    Or you can opt out like that

    ```md
    **Change log entry**

    None.
    ```
    _(possible answers No/Nope/None)_

**Comment by Strech** [comment]
:wave: Please fill "Change log entry" section in the pull request description.

  If your changes need to be present in [CHANGELOG.md](https://github.com/DataDog/dd-trace-rb/blob/master/CHANGELOG.md) you can state it this way

  ```md
  **Change log entry**

  Yes. A brief summary to be placed into the CHANGELOG.md
  ```
  _(possible answers Yes/Yep/Yeah)_

  Or you can opt out like that

  ```md
  **Change log entry**

  None.
  ```
  _(possible answers No/Nope/None)_

**Comment by Strech** [comment]
:wave: Hey @Strech, please fill "Change log entry" section in the pull request description.

If changes need to be present in [CHANGELOG.md](https://github.com/DataDog/dd-trace-rb/blob/master/CHANGELOG.md) you can state it this way

```md
**Change log entry**

Yes. A brief summary to be placed into the CHANGELOG.md
```
_(possible answers Yes/Yep/Yeah)_

Or you can opt out like that

```md
**Change log entry**

None.
```
_(possible answers No/Nope/None)_<p align="right"><i>Checked at: 2024-12-04 09:32:16
</i></p>

**Comment by Strech** [comment]
:wave: Hey @Strech, please fill "Change log entry" section in the pull request description.

If changes need to be present in [CHANGELOG.md](https://github.com/DataDog/dd-trace-rb/blob/master/CHANGELOG.md) you can state it this way

```md
**Change log entry**

Yes. A brief summary to be placed into the CHANGELOG.md
```
_(possible answers Yes/Yep/Yeah)_

Or you can opt out like that

```md
**Change log entry**

None.
```
_(possible answers No/Nope/None)_<p align="right"><i>Visited at: 2024-12-04 09:58:41</i></p>

**Comment by Strech** [comment]
:wave: Hey @Strech, please fill "Change log entry" section in the pull request description.

If changes need to be present in [CHANGELOG.md](https://github.com/DataDog/dd-trace-rb/blob/master/CHANGELOG.md) you can state it this way

```md
**Change log entry**

Yes. A brief summary to be placed into the CHANGELOG.md
```
_(possible answers Yes/Yep/Yeah)_

Or you can opt out like that

```md
**Change log entry**

None.
```
_(possible answers No/Nope/None)_<p align="right"><sup>Visited at: 2024-12-04 10:57:47 UTC</sup></p>

**Comment by Strech** [comment]
:wave: Hey @Strech, please fill "Change log entry" section in the pull request description.

If changes need to be present in [CHANGELOG.md](https://github.com/DataDog/dd-trace-rb/blob/master/CHANGELOG.md) you can state it this way

```md
**Change log entry**

Yes. A brief summary to be placed into the CHANGELOG.md
```
_(possible answers Yes/Yep/Yeah)_

Or you can opt out like that

```md
**Change log entry**

None.
```
_(possible answers No/Nope/None)_<p align="right"><sup>Visited at: 2024-12-04 11:02:48 UTC</sup></p>

**Review by y9v — APPROVED** [review]

File: .github/workflows/ensure-changelog-entry.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const options = github.rest.issues.listComments.endpoint.merge({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.payload.pull_request.number,
+            });
+            const comments = await github.paginate(options)
+            const comment = comments.find((cmnt) => {
+              return cmnt.body.startsWith("<!-- HIDDEN_MARKER -->")
```
Comment (p-datadog) [inline .github/workflows/ensure-changelog-entry.yml:49]: Maybe add "change log comment" to the "hidden marker" in case we need more than one marker?

**Review by p-datadog — APPROVED** [review]
Let's try it.

Does this also handle a comment existing between change log section header and contents (see e.g. https://github.com/DataDog/dd-trace-rb/pull/4180 for an example)?

File: .github/workflows/ensure-changelog-entry.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Strech) [reply .github/workflows/ensure-changelog-entry.yml:49]: Yeah, I had it before simplifying to hidden marker, I will rename it then 👍🏼

**Pull request merged**